### PR TITLE
feat: integrate `wasmcloud-provider-sdk`

### LINKED ISSUE #391: Integrate `wasmcloud-provider` in HTTP server provider
Once #387 is done, `wasmcloud-provider-httpsever` should be adapted to use that

### PR DESCRIPTION
**Pull request by rvolosatovs**
## Feature or Problem

This PR switches HTTP server provider to using `wasmcloud-provider-sdk`. Following merge of this PR, all subsequent providers added to the repo MUST be using `wasmcloud-provider-sdk` and *not* `wasmbus-rpc`

## Related Issues

Closes #391 

## Testing

Built on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### CODE REVIEW

File: Cargo.toml
Comment (rvolosatovs) [inline Cargo.toml:24]: These changes seem out of scope for the PR

File: crates/providers/Cargo.toml
Comment (rvolosatovs) [inline crates/providers/Cargo.toml:14]: This seems out of scope

File: Cargo.toml
Comment (thomastaylor312) [reply Cargo.toml:24]: This is applied by my toml formatter, so it is likely best practice.

File: crates/providers/Cargo.toml
Comment (thomastaylor312) [reply crates/providers/Cargo.toml:14]: Same as above

File: Cargo.toml
Comment (thomastaylor312) [reply Cargo.toml:24]: Oh and generally a lot of us other maintainers followers of what I call the "campfire" or "campground" rule which is "leave it cleaner than you found it." This was mostly unintentional because it was automatic, but I don't think it is a problem to leave it cleaner than it was

Comment (rvolosatovs) [reply Cargo.toml:24]: This change makes it very difficult to lexicographically sort the imports and copy-paste this line

File: crates/providers/Cargo.toml
Comment (rvolosatovs) [reply crates/providers/Cargo.toml:14]: This will introduce merge conflicts on each change, e.g. if two providers are added in separate PRs, whereas that would not happen with the existing approach

**Review by rvolosatovs — COMMENTED** [review]
Cannot approve PR I have filed myself, but LGTM

File: Cargo.toml
Comment (thomastaylor312) [reply Cargo.toml:24]: We can change back later if it becomes a thorn in our sides

**Review by connorsmith256 — APPROVED** [review]
LGTM

File: Cargo.toml
Comment (rvolosatovs) [reply Cargo.toml:24]: Sure, but do note that had you not made this change, there would be no merge conflict with #432 - this is primarily why I'm against out-of-scope changes in PRs and prefer PRs to be focused on doing one thing at a time - it lets us move faster with less friction

**Review by connorsmith256 — DISMISSED** [review]

**Review by rvolosatovs — COMMENTED** [review]
LGTM, good catch about the `STANDARD` vs `STANDARD_NO_PAD`, I chose the wrong engine looking at https://docs.rs/base64/latest/base64/fn.decode.html

**Review by connorsmith256 — APPROVED** [review]

**Pull request merged**
